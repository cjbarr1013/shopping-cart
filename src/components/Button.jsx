import PropTypes from 'prop-types';
import styles from '../styles/Button.module.css';

function Button({ text, handleClick, type = 'primary' }) {
  return (
    <button className={styles[type]} onClick={handleClick}>
      {text}
    </button>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  type: PropTypes.oneOf(['primary', 'secondary']),
};

export default Button;
