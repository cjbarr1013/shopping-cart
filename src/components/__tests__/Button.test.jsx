import { vi, describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '../Button';

describe('Button Component', () => {
  it('renders the button with the correct text', () => {
    render(<Button text="Click me!" handleClick={() => {}} type="primary" />);

    const button = screen.getByRole('button', { name: 'Click me!' });
    expect(button).toBeInTheDocument();
  });

  it('calls handleClick when clicked', async () => {
    const handleClick = vi.fn();
    const user = userEvent.setup();

    render(
      <Button text="Click me!" handleClick={handleClick} type="primary" />
    );

    const button = screen.getByRole('button', { name: /click me!/i });
    await user.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('does not call handleClick when not clicked', async () => {
    const handleClick = vi.fn();

    render(
      <Button text="Click me!" handleClick={handleClick} type="primary" />
    );

    expect(handleClick).not.toHaveBeenCalled();
  });

  it('applies the correct class based on the type prop', () => {
    const { rerender } = render(
      <Button text="Primary Button" handleClick={() => {}} type="primary" />
    );

    const primaryButton = screen.getByRole('button', {
      name: /primary button/i,
    });
    expect(primaryButton).toHaveClass(/primary/i);

    rerender(
      <Button text="Secondary Button" handleClick={() => {}} type="secondary" />
    );

    const secondaryButton = screen.getByRole('button', {
      name: /secondary button/i,
    });
    expect(secondaryButton).toHaveClass(/secondary/i);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Button text="Click me!" handleClick={() => {}} type="primary" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
