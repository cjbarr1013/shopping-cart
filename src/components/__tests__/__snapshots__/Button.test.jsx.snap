// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="_primary_4048ec"
  >
    Click me!
  </button>
</DocumentFragment>
`;
